components/WeekdayPicker.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
+import { useThemeColor } from '../components/Themed'
+interface Props {
+  weekdays: number[]
+  setWeekdays: React.Dispatch<React.SetStateAction<number[]>>
+}
+export default function DayPicker({ weekdays, setWeekdays }: Props) {
+  function daysIO(v: number) {
+    if (weekdays.includes(v)) {
+      const weekdayRemoved = weekdays.filter((element) => element !== v)
+      setWeekdays(weekdayRemoved)
+    } else {
+      setWeekdays([...weekdays, v])
+    }
+  }
+
+  const primaryColor = useThemeColor({}, 'primary')
+
+  const styles = StyleSheet.create({
```

Comment:
Creating styles on each render causes performance issues. Extract this outside of the component and just make style changes dynamic where necessary.